@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -198,15 +205,15 @@ export class TileDataSource<TileType extends Tile = Tile> extends DataSource {
      * `styleSetName` property) is found in `theme`.
      * @override
      */
-    setTheme(theme: Theme, languages?: string[], styleSetName?: string): void {
+    setTheme(theme: Theme | FlatTheme, languages?: string[], styleSetName?: string): void {
         if (styleSetName !== undefined) {
             this.styleSetName = styleSetName;
         }
 
-        const styleSet =
-            this.styleSetName !== undefined && theme.styles
-                ? theme.styles[this.styleSetName]
-                : undefined;
+        let styleSet: StyleSet | undefined;
+        if (this.styleSetName !== undefined) {
+            styleSet = extractStyleSet(theme, this.styleSetName);
+        }
```

Comment:
filtering the style rules here should be deprecated and considered only as a temporary solution until we are with the deprecation of the old theme format.

Our goal should be to deprecate (and eventually remove) the concept that there is only one single "styleSet" associated to a datasource. That is, currently the datasources must set the property `"styleSetName"` at construction type and then later this property is used to filter the style rules. This is not how the unified theme is supposed to work.

In the unified theme format the `styleSet` is just a way to group together style rules, for example a styleSet called "roads" could be used to group layers for the roads, another one called city-center could collect the layers for the city centers and so on.. These groups of rules could later be references in `theme.priorities` and `theme.labelPriorities` to source items. Eventually we could also have APIs to enable/disable these style sets.

Reply:
I added your comment to the ticket I created for the full FlatTheme support: HARP-12835